core/src/test/java/dev/failsafe/functional/BulkheadTest.java

```diff
@@ -46,6 +51,27 @@ public void testPermitAcquiredAfterWait() {
     }, "test");
   }
 
+  public void testPermitAcquiredAfterWaitWithLargeQueue(){
+    Bulkhead<Object> bulkhead = Bulkhead.builder(1).withMaxWaitTime(Duration.ofSeconds(15)).build();
+    FailsafeExecutor<Object> exec = Failsafe.with(bulkhead);
+    CompletableFuture<Void>[] tasks = new CompletableFuture[10];
+    for(int i = 0; i < tasks.length; i++){
+      int index = i;
+      CheckedRunnable sleep = () -> {
+        ignoreExceptions(() ->{
+          System.out.println("Running sleep task " + (index + 1));
+          TimeUnit.SECONDS.sleep(1);
```

Comment:
Thanks for this! One suggestion: can you tweak the test case so it runs faster? Maybe a 10ms sleep, where the entire test case runs in < 250ms if possible.

Reply:
Done.